Projects admins can't add groups to projects when using LDAP

We are using Harbor 2.3.0 together with LDAP for authentication. We are finding that project admins can't add groups to projects and only Harbor admins can add groups. 
When a project admin clicks the group button only a spinning circle appears as if there is no response or time out. When performing the same procedure as a harbor admin the popup works as expected.

<img width="858" alt="Screenshot 2021-08-16 at 09 31 44" src="https://user-images.githubusercontent.com/8925667/129527281-894a3777-ac34-4d5d-99fb-dd2ad3aaba32.png">

Looking at the browsers developer console we can see that a call to /usergroup fails with a 403.
Manually calling the usergroup function from the API-helper results in the same 403.
 
We are not seeing any errors in the Harbor logs, or any kind of logging related to the action.
Is this by design or is it a possible misconfiguration on our side?

@stonezdj I am getting a 500 reply to the POST on `api/v2.0/projects/<id>/members` and the harbor log has: 
`2021-08-17T07:15:17Z [ERROR] [/lib/http/error.go:54]: {"errors":[{"code":"UNKNOWN","message":"unknown: a LDAP user group with same DN already exist"}]}`
Harbor version v2.3.0-047b122c. I am an admin user through a dedicated `LDAP Group Admin DN`.

Possibly this should be a separate issue altogether and is not just related to error message not propagating to the UI. Users should be able to add Group membership via explicit DN even if the Group has already been ingested into Harbor.

Regardless of these issues, Harbor should have functionality to allow non-admin users or at least users with Project Admin role within a project they created to add Project Admin Group membership. Not having this ability creates a huge bottleneck for admin users who should be focusing on managing overall Harbor configurations instead of micromanaging hundreds of individual project. Maybe this should be a separate issue as well, please advise.

--- a/src/portal/src/app/base/project/member/member.component.html
+++ b/src/portal/src/app/base/project/member/member.component.html
@@ -16,7 +16,7 @@
             <button class="btn  btn-secondary" (click)="openAddMemberModal()" [disabled]="!hasCreateMemberPermission">
                 <span><clr-icon shape="plus" size="16"></clr-icon>&nbsp;{{'MEMBER.USER' | translate }}</span>
             </button>
-            <button class="btn  btn-secondary" (click)="openAddGroupModal()" [disabled]="!hasCreateMemberPermission || !(isLdapMode || isHttpAuthMode || isOidcMode)">
+            <button class="btn  btn-secondary" (click)="openAddGroupModal()" [disabled]="!hasCreateMemberPermission || !((isLdapMode && currentUser?.has_admin_role) || isHttpAuthMode || isOidcMode)">
                 <span><clr-icon shape="plus" size="16"></clr-icon>&nbsp;{{'MEMBER.LDAP_GROUP' | translate}}</span>
             </button>
             <clr-dropdown [clrCloseMenuOnItemClick]="false" class="btn  btn-link" clrDropdownTrigger>

--- a/src/portal/src/app/base/project/member/member.component.spec.ts
+++ b/src/portal/src/app/base/project/member/member.component.spec.ts
@@ -4,7 +4,7 @@ import { CUSTOM_ELEMENTS_SCHEMA } from '@angular/core';
 import { ActivatedRoute } from "@angular/router";
 import { MessageHandlerService } from "../../../shared/services/message-handler.service";
 import { SessionService } from "../../../shared/services/session.service";
-import { MemberService } from "./member.service";
+import { MemberService } from "../../../../../ng-swagger-gen/services/member.service";
 import { AppConfigService } from "../../../services/app-config.service";
 import { of } from 'rxjs';
 import { OperationService } from "../../../shared/components/operation/operation.service";
@@ -14,20 +14,42 @@ import { ConfirmationDialogService } from "../../global-confirmation-dialog/conf
 import { SharedTestingModule } from "../../../shared/shared.module";
 import { HttpHeaders, HttpResponse } from "@angular/common/http";
 import { Registry } from "../../../../../ng-swagger-gen/models/registry";
+import { ProjectMemberEntity } from "../../../../../ng-swagger-gen/models/project-member-entity";
+import { delay } from "rxjs/operators";
 
 describe('MemberComponent', () => {
     let component: MemberComponent;
     let fixture: ComponentFixture<MemberComponent>;
+    const mockedMembers: ProjectMemberEntity[] = [
+        {
+            id: 1,
+            project_id: 1,
+            entity_name: 'test1',
+            role_name: 'projectAdmin',
+            role_id: 1,
+            entity_id: 1,
+            entity_type: 'u'
+        },
+        {
+            id: 2,
+            project_id: 1,
+            entity_name: 'test2',
+            role_name: 'projectAdmin',
+            role_id: 1,
+            entity_id: 2,
+            entity_type: 'u'
+        }
+    ];
     const mockMemberService = {
         getUsersNameList: () => {
             return of([]);
         },
         listProjectMembersResponse: () => {
             const response: HttpResponse<Array<Registry>> = new HttpResponse<Array<Registry>>({
-                headers: new HttpHeaders({'x-total-count': '0'}),
-                body: []
+                headers: new HttpHeaders({'x-total-count': '2'}),
+                body: mockedMembers
             });
-            return of(response);
+            return of(response).pipe(delay(0));
         },
         updateProjectMember: () => {
             return of(null);
@@ -118,11 +140,16 @@ describe('MemberComponent', () => {
     beforeEach(() => {
         fixture = TestBed.createComponent(MemberComponent);
         component = fixture.componentInstance;
-        component.loading = true;
         fixture.detectChanges();
     });
 
     it('should create', () => {
         expect(component).toBeTruthy();
+    });
+    it('should render member list', async () => {
+        fixture.autoDetectChanges(true);
+        await fixture.whenStable();
+        const rows = fixture.nativeElement.querySelectorAll('clr-dg-row');
+        expect(rows.length).toEqual(2);
     });
 });

--- a/src/portal/src/app/base/project/member/member.component.ts
+++ b/src/portal/src/app/base/project/member/member.component.ts
@@ -55,7 +55,7 @@ export class MemberComponent implements OnInit, OnDestroy {
   roleNum: number;
   isDelete = false;
   isChangeRole = false;
-  loading = false;
+  loading = true;
 
   isChangingRole = false;
   batchChangeRoleInfos = {};

